src/basilisp/core.lpy

```diff
@@ -4735,7 +4735,12 @@
 
 (defn alter-var-root
   "Atomically alter the Var root by calling ``(apply f root args)`` and setting the root
-  as the result."
+  as the result. Returns the new value.
+
+  Note: Due to Basilisp's Direct Linking Optimization, changes to a
+  Var's root value may not be reflected in the code unless the Var is
+  dynamic. To ensure updates propagate, set the Var's `^:redef`
+  metadata, which disables direct linking."
```

Comment:
```suggestion
  .. note::
  
     Due to Basilisp's :ref:`Direct Linking Optimization <direct_linking>`, changes to
     a Var's root value may not be reflected in the code unless the Var is dynamic.
     To ensure updates propagate, set the Var's `^:redef` metadata, which disables
     direct linking."
```

Just a change to the markup to show the note in a little bit more clear way:

<img width="211" alt="image" src="https://github.com/user-attachments/assets/c0b91b30-4842-45f3-bda4-7260adbcba62">

Also added a link to the main direct linking section.